gammapy/scripts/analysis.py

```diff
@@ -44,24 +44,25 @@ class Analysis:
     ----------
     config : dict
         Configuration options following `Config` schema
-    template : str
-        Template for configuration settings
 
     Examples
     --------
     Example how to create an Analysis object:
 
     >>> from gammapy.scripts import Analysis
-    >>> analysis = Analysis(template="1d")
+    >>> analysis = Analysis.from_template(template="1d")
 
     TODO: show a working example of running an analysis.
     Probably not here, but in high-level docs, linked to from class docstring.
     """
 
-    def __init__(self, config=None, template="basic"):
-        self._config = Config(config, template)
```

Comment:
This is a relevant point.
I think it's good to have a default settings by default when instantiating `Analysis`, if not I think we risk having validation errors during the analysis process.

Reply:
I see your point. I guess in the end it's just a matter of the API. As far as I understood these two things would lead to the same result :
```
# old
analysis = Analysis(config, template="basic")

# new
analysis = Analysis.from_template("basic")
analysis.config.update_settings(config)
```
Is this correct?

If so I find the latter solution a little bit simpler and more explicit, because it's clear, that you start from a basic configuration and then overwrite and extend it by user settings.
In the first case it was unclear to me at first how the `template` and the `config` are combined internally. It also lead to much more complicated code, because you had to pass the `template` info to the `Config` object, where it was read from file and then the user and template setting were stored and handled separately.

Reply:
Ah, I guess the difference is that in the first case the `config` is validated on init, while in the second case `config` is not validated?

I guess it's probably a good idea if the validation always happens on `.update_settings()` anyway?

Reply:
I agree with you.
But what I meant was just having basic default settings when doing `analysis = Analysis()` which I guess it is not forbidden.

Reply:
I could also restore the behavior, that `Analysis.from_template("basic")` is equivalent to `Analysis(config=None)` if you prefer. Alternatively we can make `config` a required
 argument...

Reply:
> I could also restore the behavior, that Analysis.from_template("basic") is equivalent to Analysis(config=None)

Yep
If the user really wants an empty config it's just `Analysis(config={})` should do it, no?

Reply:
OK, I'll change the init default.

Reply:
Done